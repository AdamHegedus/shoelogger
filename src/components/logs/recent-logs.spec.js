import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import RecentLogs from '@/components/logs/recent-logs';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('recent-logs.vue', () => {
    describe('meta.status is true', () => {
        it('should match snapshot', () => {
            // ASSIGN
            const logsGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 3,
                        timestamp: 100
                    },
                    {
                        id: 3,
                        product: 'Bar',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 4,
                        product: 'Bar',
                        brand: 'foobar',
                        distance: 5,
                        timestamp: 100
                    },
                    {
                        id: 5,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 6,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 5,
                        timestamp: 100
                    },
                    {
                        id: 7,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 5,
                        timestamp: 100
                    },
                    {
                        id: 8,
                        product: 'Bar',
                        brand: 'foobar',
                        distance: 5,
                        timestamp: 100
                    },
                    {
                        id: 9,
                        product: 'bar',
                        brand: 'foobar',
                        distance: 10,
                        timestamp: 100
                    },
                    {
                        id: 10,
                        product: 'Bar',
                        brand: 'foobar',
                        distance: 50,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        getters: logsGetters,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(RecentLogs, { store, localVue });
            wrapper.vm.selectLog(0);

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
            expect(navigationActions.setNavigation.mock.calls[0][1]).toMatchSnapshot();
        });
    });

    describe('meta.status is false', () => {
        it('should match snapshot', () => {
            // ASSIGN
            const logsGetters = {
                getMeta: () => {
                    return {
                        status: false,
                        message: 'Error message text'
                    };
                }
            };

            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 3,
                        timestamp: 100
                    },
                    {
                        id: 3,
                        product: 'Bar',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 4,
                        product: 'Bar',
                        brand: 'foobar',
                        distance: 5,
                        timestamp: 100
                    },
                    {
                        id: 5,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 6,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 5,
                        timestamp: 100
                    },
                    {
                        id: 7,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 5,
                        timestamp: 100
                    },
                    {
                        id: 8,
                        product: 'Bar',
                        brand: 'foobar',
                        distance: 5,
                        timestamp: 100
                    },
                    {
                        id: 9,
                        product: 'bar',
                        brand: 'foobar',
                        distance: 10,
                        timestamp: 100
                    },
                    {
                        id: 10,
                        product: 'Bar',
                        brand: 'foobar',
                        distance: 50,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        getters: logsGetters,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(RecentLogs, { store, localVue });
            wrapper.vm.selectLog(0);

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
            expect(navigationActions.setNavigation.mock.calls[0][1]).toMatchSnapshot();
        });
    });


    describe('selectLog', () => {
        it('should set index when selectedIndex is unset', () => {
            // ASSIGN
            const selectedIndex = 0;

            const logsGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 3,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        getters: logsGetters,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(RecentLogs, { store, localVue });

            // ACT
            wrapper.vm.selectLog(selectedIndex);

            // ASSERT
            const actual = wrapper.vm.selectedIndex;
            expect(actual).toMatchSnapshot();
        });

        it('should set index when selectedIndex is the same', () => {
            // ASSIGN
            const selectedIndex = 0;
            const originalIndex = 0;

            const logsGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 3,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        getters: logsGetters,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(RecentLogs, { store, localVue });
            wrapper.setData({
                selectedIndex: originalIndex
            });

            // ACT
            wrapper.vm.selectLog(selectedIndex);

            // ASSERT
            const actual = wrapper.vm.selectedIndex;
            expect(actual).toMatchSnapshot();
        });

        it('should set index when selectedIndex is different', () => {
            // ASSIGN
            const selectedIndex = 1;
            const originalIndex = 0;

            const logsGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 3,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        getters: logsGetters,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(RecentLogs, { store, localVue });
            wrapper.setData({
                selectedIndex: originalIndex
            });

            // ACT
            wrapper.vm.selectLog(selectedIndex);

            // ASSERT
            const actual = wrapper.vm.selectedIndex;
            expect(actual).toMatchSnapshot();
        });

        it('should call resetMeta when meta.status is false', () => {
            // ASSIGN
            const selectedIndex = 0;
            const logsGetters = {
                getMeta: () => {
                    return {
                        status: false,
                        message: null
                    };
                }
            };

            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 3,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        getters: logsGetters,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(RecentLogs, { store, localVue });
            // ACT
            wrapper.vm.selectLog(selectedIndex);

            // ASSERT
            const actual = logsActions.resetMeta;
            expect(actual).toBeCalled();
        });


        it('should not call resetMeta when meta.status is true', () => {
            // ASSIGN
            const selectedIndex = 0;

            const logsGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 3,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        getters: logsGetters,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(RecentLogs, { store, localVue });

            // ACT
            wrapper.vm.selectLog(selectedIndex);

            // ASSERT
            const actual = logsActions.resetMeta;
            expect(actual).not.toBeCalled();
        });
    });

    describe('deleteLog', () => {
        it('should call event', () => {
            // ASSIGN
            const logsGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Foo',
                        brand: 'foobar',
                        distance: 3,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        getters: logsGetters,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(RecentLogs, { store, localVue });
            wrapper.vm.selectLog(0);

            // ACT
            wrapper.find('button.log-delete').trigger('click');

            // ASSERT
            expect(logsActions.deleteLog.mock.calls.length).toEqual(1);
            expect(logsActions.deleteLog.mock.calls[0][1]).toMatchSnapshot();
        });
    });
});
