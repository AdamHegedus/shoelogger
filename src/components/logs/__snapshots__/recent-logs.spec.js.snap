// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recent-logs.vue deleteLog should call event 1`] = `
Object {
  "id": 0,
}
`;

exports[`recent-logs.vue meta.status is false should match snapshot 1`] = `
<div>
  <h3>
    Recent logs
  </h3>
   
  <span
    class="row"
    name="list"
    tag="div"
  >
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card selected"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              1
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style=""
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <p
        appear=""
        class="alert alert-danger mt-3"
        name="animation-slide-bottom-and-fade"
      >
        
                    Error message text
                
      </p>
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              1
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              3
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Bar
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              1
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Bar
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              5
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              1
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              5
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              5
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Bar
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              5
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        bar
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              10
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
  </span>
</div>
`;

exports[`recent-logs.vue meta.status is false should match snapshot 2`] = `"/logs/recent"`;

exports[`recent-logs.vue meta.status is true should match snapshot 1`] = `
<div>
  <h3>
    Recent logs
  </h3>
   
  <span
    class="row"
    name="list"
    tag="div"
  >
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card selected"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              1
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style=""
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              1
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              3
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Bar
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              1
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Bar
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              5
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              1
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              5
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Foo
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              5
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        Bar
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              5
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
    <div
      class="col-12 col-md-6 col-lg-6 px-3 px-lg-2 py-2 transition-item"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            
                        bar
                        
            <br />
             
            <small>
              foobar
            </small>
          </h5>
           
          <span
            class="distance-text"
          >
            
                        Distance:
                        
            <span
              class="distance-value"
            >
              10
            </span>
            
                        km
                    
          </span>
           
          <span
            class="float-right"
          >
            
          </span>
           
          <span
            class="timestamp"
          >
            Last modified: 100
          </span>
           
          <button
            appear=""
            class="btn btn-danger log-action log-delete"
            mode="out-in"
            name="fade"
            style="display: none;"
          >
            
                            Delete
                        
          </button>
        </div>
      </div>
       
      <!---->
    </div>
  </span>
</div>
`;

exports[`recent-logs.vue meta.status is true should match snapshot 2`] = `"/logs/recent"`;

exports[`recent-logs.vue selectLog should set index when selectedIndex is different 1`] = `1`;

exports[`recent-logs.vue selectLog should set index when selectedIndex is the same 1`] = `null`;

exports[`recent-logs.vue selectLog should set index when selectedIndex is unset 1`] = `0`;
